app/models/case_log.rb

```diff
@@ -213,12 +213,33 @@ def update_status!
                   end
   end
 
+  def reset_not_routed_questions
+    form.invalidated_page_questions(self).each do |question|
+      enabled = form.enabled_page_questions(self)
+      contains_selected_answer_option = enabled.map(&:id).include?(question.id) && enabled.find { |q| q.id == question.id }.answer_options.values.map { |x| x["value"] }.include?(public_send(question.id))
+      unless contains_selected_answer_option
+        public_send("#{question.id}=", nil) if respond_to?(question.id.to_s)
+      end
+    end
+  end
+
+  def reset_derived_questions
+    dependent_questions = { layear: [{ key: :renewal, value: "No" }], homeless: [{ key: :renewal, value: "No" }] }
```

Comment:
Can we derive this from the JSON rather than hardcoding it?

Reply:
We don't really have anything that indicates that these are inferred in the JSON. Would it make sense to add it just for this reason?

Reply:
hmm we have:

```
"layear": {
    "depends_on":  [{ "renewal": "No" }]
},
"homeless": {
    "depends_on":  [{ "renewal": "No" }]
}
```

If we collect those and apply this does it break for other fields?

Reply:
So we're already getting the depends on to determine if something is not routed to and if it's not we clear it.
In this case we want to clear something that is routed to, but only if it's one of the fields we're inferring. We can get the questions that are routed to from this JSON, but we would still need to filter them down to the ones that get inferred, no?

Reply:
Otherwise we'd just clear everything?

Reply:
Ah yeah I remember now.  :+1: